include/nbl/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.h

```diff
@@ -22,15 +22,32 @@ namespace nbl::asset::material_compiler
 // TODO: we need a GLSL to C++ compatibility wrapper
 #define uint uint32_t
 #define uvec2 uint64_t
+struct nbl_glsl_MC_emitter_t
+{
+	uvec2 emissive;
+	uvec2 emissionProfile;
+	float normalizeEnergy;
+	float orientation[6]; // not vector because of alignment, [0:3] left [3:6] up
+};
 struct nbl_glsl_MC_oriented_material_t
 {
-    uvec2 emissive;
+	uint emitter_id; // 0 = invalid
     uint prefetch_offset;
     uint prefetch_count;
     uint instr_offset;
     uint rem_pdf_count;
     uint nprecomp_count;
     uint genchoice_count;
```

Comment:
maybe pull in the GLSL structs (so the fields are in sync and don't repeat) and use inheritance to extend them with extra methods